### PR TITLE
Add support for Password Hardening

### PR DESCRIPTION
**Pull request by davidpil2002**
### Description of PR

Summary:
Added Password Hardening Tests according to HLD Test Plan

Context: By enabling the new Password Hardening feature and configured policies, the user will be enforce to use "strong" passwords according to the policies configured.

The new tests cases will cover all the possible password policies with good and bad cases, including:
* Perform show command
* Perform config command
* Verify default values of the feature
* Enable/Disable the feature
* Test all passw policies available:
  * Configure passw policy classes of every existing types
  * Configure passw policy  min using different valid values
  * Configure passw policy  age expiration using different valid values
  * Configure passw policy  age warning using different valid values
  * Configure passw policy  username-password-match
  * Configure passw policy  history using different valid values
Note
After creating a new policy is necessary to set a new password for a user to verify that the policy matches the configured.
The set configuration should be validated using the show command.
### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [*] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
Add new tests cases for testing the feature Password Hardening.

#### How did you do it?
There are 10 test scenarios each one of them config different policies and test the case with good(strong) passwords and bad(weak) passwords expecting results according to the configuration and the flow.
The configuration is done by using sonic-cli, sending different "config" commands.
In addition, the test is testing that the configuration in PAM files in the system change according to the expected.
The expected files were saved in the directory "sample".

#### How did you verify/test it?
New tests cases passed
#### Any platform specific information?
N/A
#### Supported testbed topology if it's a new test case?
any
### Documentation

Link of HLD including feature design and Test plan:
https://github.com/Azure/SONiC/blob/master/doc/passw_hardening/hld_password_hardening.md

### CODE REVIEW

File: tests/passw_hardening/test_passw_hardening.py
Comment (liuh-80) [inline tests/passw_hardening/test_passw_hardening.py:605]: I have a question for this scenario, may be need a UT to cover it:

let's say the device only have 1 user ‘admin’, and then admin password expired, so is there any solution to login to device?

If not this will cause very big incident, we need change feature to prevent this happen, for example add a white list or make su user not limit by this feature.

File: tests/passw_hardening/test_passw_hardening.py
Comment (davidpil2002) [reply tests/passw_hardening/test_passw_hardening.py:286]: done

File: tests/passw_hardening/test_passw_hardening.py
Comment (davidpil2002) [reply tests/passw_hardening/test_passw_hardening.py:605]: this case its not really related to password-hardening feature.
If someone forget the password there are different ways to recover from it and it could happen even without the feature.
The admin should know how to support it.

File: tests/passw_hardening/test_passw_hardening.py
Comment (davidpil2002) [reply tests/passw_hardening/test_passw_hardening.py:286]: (I used pytest.skip, because pytest_require raise some problems to me)

File: tests/passw_hardening/test_passw_hardening.py
Comment (davidpil2002) [reply tests/passw_hardening/test_passw_hardening.py:269]: this feature should not block your login from remote authentication.
By design, PAM lib handles remote and local authentication separately. So local password policies are applied to local authentications.

**Comment by liat-grozovik** [comment]
@liuh-80 could you please review and signoff?
@davidpil2002  should we wait for the merge once the feature is merged?

**Comment by davidpil2002** [comment]
> @liuh-80 could you please review and signoff? @davidpil2002 should we wait for the merge once the feature is merged?

yes, first the feature from sonic-buildimage should be merged, then the sonic-mgmt code

**Review by liuh-80 — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@roysr-nv please also review

**Comment by roysr-nv** [comment]
> @roysr-nv please also review

it was reviewed by @AntonHryshchuk

**Comment by zhangyanzhao** [comment]
@davidpil2002 can you please help to sign  the EasyCLA to unblock the merge? Thanks @liat-grozovik

**Pull request merged**
